yaramo/route.py

```diff
@@ -56,6 +56,31 @@ def duplicate(self):
         new_obj.end_signal = self.end_signal
         return new_obj
 
+    def update_maximum_speed(self):
+        maximum_speed = self.start_signal.edge.maximum_speed or float('inf')
+        edges = self.get_edges_in_order()
+        nodes = []
+
+        previous_node = self.start_signal.next_node()
+        nodes.append(None)
+        nodes.append(previous_node)
+
+        for edge in edges[1:]:
+            next_node = edge.node_a if edge.node_b == previous_node else edge.node_b
+            nodes.append(next_node)
+            previous_node = next_node
+
+        # We have to look back and ahead, as we have to figure out which branch of the node we traverse
+        for ((previous_node, node), next_node), edge in zip(zip(zip(nodes, nodes[1:]), nodes[2:]), edges[1:]):
```

Comment:
This might be one of the few cases in Python where using indices for iteration could be easier to read. Otherwise, comments for the if statements could also help make it easier to understand.